xunit.runner.visualstudio.testadapter/Utility/RunSettingsHelper.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+#if !PLATFORM_DOTNET
+    using System.Xml;
+#endif
+
+namespace Xunit.Runner.VisualStudio.TestAdapter
+{
+    internal static class RunSettingsHelper
+    {
+        public static bool DisableAppDomain { get; set; }
+
+        public static bool DisableParallelization { get; set; }
+
+        /// <summary>
+        /// Reads settings for the current run from run context
+        /// </summary>
+        /// <param name="runContext">RunContext of the run</param>
+        public static void ReadRunSettings(IRunContext runContext)
+        {
+            var settingsXml = runContext?.RunSettings?.SettingsXml;
+
+#if !PLATFORM_DOTNET
+            if (!string.IsNullOrEmpty(settingsXml))
+            {
+                try
+                {
+                    using (var stringReader = new StringReader(settingsXml))
+                    {
+                        using (var xmlReader = XmlReader.Create(stringReader, new XmlReaderSettings { IgnoreComments = true, IgnoreWhitespace = true }))
+                        {
+                            if (ReadToRunConfigurationElement(xmlReader))
+                            {
+                                // Process the fields in Xml elements
+                                xmlReader.Read();
+                                var numSettingsRead = 0;
+                                while (xmlReader.NodeType == XmlNodeType.Element && numSettingsRead < 2)
+                                {
+                                    string elementName = xmlReader.Name;
```

Comment:
I would recommend using XPath or XLinq (System.Xml.Linq.XDocument) to parse the xml since I think that would be easier, less error prone and also more future proof.

With XLinq you could write something like below code to parse the xml -

```
            var settingsXml = @"<?xml version=""1.0""?>
<RunSettings>
    <RunConfiguration>
        <TargetPlatform>x86</TargetPlatform>
        <DisableAppDomain>true</DisableAppDomain>
        <DisableParallelization>true</DisableParallelization>
    </RunConfiguration>
</RunSettings>";

            var doc = XDocument.Parse(settingsXml);

            var disableAppDomainString = doc?.Element("RunSettings")?.Element("RunConfiguration")?.Element("DisableAppDomain")?.Value;
            bool disableAppDomain;
            bool.TryParse(disableAppDomainString, out disableAppDomain);
            Console.WriteLine(disableAppDomain);

            var disableParallelizationString = doc?.Element("RunSettings")?.Element("RunConfiguration")?.Element("DisableParallelization")?.Value;
            bool disableParallelization;
            bool.TryParse(disableParallelizationString, out disableParallelization);
            Console.WriteLine(disableParallelization);

            var blahString = doc?.Element("RunSettings")?.Element("RunConfiguration")?.Element("Blah")?.Value;
            bool blah;
            bool.TryParse(blahString, out blah);
            Console.WriteLine(blah);
```

The above code should print the following output -
```
True
True
False
```